test/integration/connections/test_bucketfs_location.py

```diff
@@ -1,86 +1,177 @@
-    validate_bfs_path(bfs_path)
 
 
-def test_create_bucketfs_location_saas_db_name(use_saas,
-                                               saas_host,
-                                               saas_pat,
-                                               saas_account_id,
-                                               backend_aware_saas_database_id,
-                                               database_name):
-    if not use_saas:
-        pytest.skip("The test is not configured to use SaaS.")
-
-    kwargs = {
-        StdParams.saas_url.name: saas_host,
-        StdParams.saas_account_id.name: saas_account_id,
-        StdParams.saas_database_name.name: database_name,
-        StdParams.saas_token.name: saas_pat,
-        StdParams.path_in_bucket.name: 'test_path'
-    }
-    bfs_path = create_bucketfs_location(**kwargs)
-    validate_bfs_path(bfs_path)
+@pytest.fixture(scope='session')
+def saas_params_with_name(saas_params_with_id,
+                          database_name) -> dict[str, Any]:
+    saas_params = dict(saas_params_with_id)
+    saas_params.pop(StdParams.saas_database_id.name)
+    saas_params[StdParams.saas_database_name.name] = database_name
+    return saas_params
 
 
-def test_create_bucketfs_location_error():
-    kwargs = {
+@pytest.fixture
+def rubbish_params() -> dict[str, Any]:
+    return {
         StdParams.bucketfs_name.name: 'bfsdefault',
         StdParams.bucket.name: 'default',
         StdParams.saas_url.name: 'my_saas_url',
     }
+
+
+@contextmanager
+def write_test_file(bfs_path: bfs.path.PathLike) -> None:
+    bfs_path.write(TEST_FILE_CONTENT)
+    try:
+        yield
+    finally:
+        # We cannot reuse the same path, in subsequent tests because of the
+        # temporary lock bucket-fs places on deleted files, but it's useful
+        # to delete the file anyway to avoid potential false-positives.
+        bfs_path.rm()
+
+
+def validate_test_file(bfs_path: bfs.path.PathLike) -> None:
+    file_content = b''.join(bfs_path.read())
+    assert file_content == TEST_FILE_CONTENT
+
+
+def validate_conn_object(pyexasol_connection: pyexasol.ExaConnection,
+                         conn_name: str,
+                         conn_obj: ConnObjectData):
+    bfs_path = create_bucketfs_location_from_conn_object(conn_obj)
+    validate_test_file(bfs_path)
+
+
+def test_create_bucketfs_location_onprem(use_onprem,
+                                         onprem_params):
+    if not use_onprem:
+        pytest.skip("The test is not configured to use ITDE.")
+
+    extra_params = {StdParams.path_in_bucket.name: 'test_create_location'}
+    bfs_path = create_bucketfs_location(**onprem_params, **extra_params)
+    with write_test_file(bfs_path):
+        validate_test_file(bfs_path)
+
+
+def test_create_bucketfs_location_saas_db_id(use_saas,
+                                             saas_params_with_id):
+    if not use_saas:
+        pytest.skip("The test is not configured to use SaaS.")
```

Comment:
Is it possible to achieve this decorator based?

Reply:
Possibly, but it's not trivial, as we need to look at the fixture.